### PR TITLE
expose machine type in API & default to q35

### PR DESCRIPTION
**Pull request by mpolednik**
Currently, KubeVirt doesn't make conscious decision about default machine type or let the user specify the type explicitly. This PR aims to change both by making `q35` the default machine type (which is just an alias for newest machine type, e.g. q35=pc-q35-2.10) and allowing user to explicitly request a machine type.

The reasoning why q35 should be the default instead of i440fx is that in QEMU, it's considered the modern architecture; but it's important to remember that just changing the machine type doesn't make everything magically better: in the future, we need to think about default device models that play nicely with PCIe, such as qemu-xhci for USB.

### CODE REVIEW

**Comment by mpolednik** [comment]
oVirt standard CI seems to have wrong dependencies:

```
# kubevirt.io/kubevirt/vendor/k8s.io/apimachinery/pkg/util/strategicpatch
../vendor/k8s.io/apimachinery/pkg/util/strategicpatch/patch.go:520: undefined: sort.SliceStable
# kubevirt.io/kubevirt/vendor/k8s.io/apimachinery/pkg/util/httpstream/spdy
../vendor/k8s.io/apimachinery/pkg/util/httpstream/spdy/roundtripper.go:166: tlsConfig.Clone undefined (type *tls.Config has no field or method Clone, but does have tls.clone)
make: *** [compile] Error 1
```

Travis failure seems to be caused by unavailable `updates` repo:

```
Error: Failed to synchronize cache for repo 'updates'
```

Could someone trigger re-run of the tests? I've just tested that the code is working, and it should also be tested (API by converter_tests.go changes, defaults by new default_test.go under virtwrap/api).

**Comment by rmohr** [comment]
retest this please

**Comment by rmohr** [comment]
ok to test

**Comment by rmohr** [comment]
#655 should resolve these issues:

```bash
./hack/build-go.sh install 
# kubevirt.io/kubevirt/vendor/k8s.io/apimachinery/pkg/util/strategicpatch
../vendor/k8s.io/apimachinery/pkg/util/strategicpatch/patch.go:520: undefined: sort.SliceStable
# kubevirt.io/kubevirt/vendor/k8s.io/apimachinery/pkg/util/httpstream/spdy
../vendor/k8s.io/apimachinery/pkg/util/httpstream/spdy/roundtripper.go:166: tlsConfig.Clone undefined (type *tls.Config has no field or method Clone, but does have tls.clone)
make: *** [Makefile:29: compile] Error 2
```

**Comment by rmohr** [comment]
retest this please

**Comment by rmohr** [comment]
retest this please

File: pkg/api/v1/schema.go [hunk 1]
```diff
@@ -65,6 +68,11 @@ type ResourceRequirements struct {
 	Requests v1.ResourceList `json:"requests,omitempty"`
 }
 
+type Machine struct {
```
Comment (rmohr) [inline pkg/api/v1/schema.go]: Are there other attributes which would be later placed in this section?

**Comment by rmohr** [comment]
retest this please

**Comment by rmohr** [comment]
retest this please

**Comment by rmohr** [comment]
@mpolednik I have the  suspicion that there is really an issue with the vm configuration. At first I thought it is just CI, because I also had issues with #638, but also there it turned out that the vm configuration was the issue.

In your case these tests

```bash
Summarizing 3 Failures:

[Fail] CloudInit UserData CloudInit Data Source NoCloud [It] should launch vm with cloud-init data source NoCloud 
/root/go/src/kubevirt.io/kubevirt/tests/vm_userdata_test.go:65

[Fail] CloudInit UserData CloudInit Data Source NoCloud [It] should launch ephemeral vm with cloud-init data source NoCloud 
/root/go/src/kubevirt.io/kubevirt/tests/vm_userdata_test.go:65

[Fail] CloudInit UserData CloudInit Data Source NoCloud [It] should launch VMs with user-data in k8s secret 
/root/go/src/kubevirt.io/kubevirt/tests/vm_userdata_test.go:65
```

**Comment by mpolednik** [comment]
@rmohr Will look into that more closely tomorrow - could be some side effect of the defaults.

**Comment by mpolednik** [comment]
It seems that there is an issue with cirros & q35 with default libvirt domain, let's investigate that before going further.

**Comment by mpolednik** [comment]
retest this please

**Pull request merged**